app/models/blog.rb

```diff
@@ -3,6 +3,9 @@ class Blog < ApplicationRecord
   has_many :comments
   has_many :likes
   after_save :update_post
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :commentsCounter, numericality: { greater_than_or_equal_to: 0 }
+  validates :likesCounter, numericality: { greater_than_or_equal_to: 0 }
 
```

Comment:
- [ ] The _index_ and _show_ action of your _Blogs Controller (app\controllers\blogs_controller.rb)_  should have query similar to: `@posts = @user.blog.includes(:comments)` instead of `@blog = @user.blog`. Doing so solves N+1 problem in your code.  Kindly ensure to improve upon this aspect.

Reply:
Thank you @rloterh